lib/pure/collections/setimpl.nim

```diff
@@ -69,6 +68,11 @@ template containsOrInclImpl() {.dirty.} =
     rawInsert(s, s.data, key, hc, -1 - index)
     inc(s.counter)
 
+template doWhile(a, b) =
+  while true:
+    b
+    if not a: break
```

Comment:
I think this should be inlined as its only used once and its name doesn't really reflect what it does (runs b once even if a is false from the start)

Reply:
I don't really disagree.  In fact I inlined it just as you say in my https://github.com/c-blake/adix package (e.g. in `lpset.nim`).

I left it that way because the original code being restored was that way.  So, minimizing the pre-Tombstone to post-restored-no-Tombstone diff.

It is named after the C do-while construct, but because of indentation based block structure there's not really a better way to name it, IMO.  I agree it could go, but that should maybe be a follow-on clean-up PR.

Reply:
I think its fine to inline it in this PR :)

Reply:
Let's do one thing at a time for simplicity.  I have a couple clean-ups and API enhancements in mind.

Reply:
While I have your attention, do you think `getCap` is a good name for the `data.len`?  `len` itself is already used for `count`.  I have been using `getCap` and `setCap` in https://github.com/c-blake/adix.  Both are kind of missing by oversight in both `sets` and `tables`.

Reply:
I think the names are fine. Its kind of analog to seq's capacity.